ICE: No kind for cast

The following only ICE's on stable, beta and nightly:

```Rust
fn main() {
    let f = [();  { &loop { break } as *const _ as usize } ];
}
```
playground: https://play.rust-lang.org/?gist=b35ac8a560e0cc448e95e2a62598208a&version=stable&mode=release&edition=2015
Backtrace (stable):
```
error[E0019]: constant contains unimplemented expression type
   --> src\lib.rs:164:26
    |
164 |         let f = [();  { &loop { break } as *const _ as usize } ];
    |                          ^^^^^^^^^^^^^^

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: EvalError { kind: a raw memory access tried to access part of a pointer value as raw bytes, backtrace: None }', libcore\result.rs:945:5
stack backtrace:
   0: <u128 as compiler_builtins::int::Int>::min_value
   1: core::alloc::<impl core::alloc::::Opaque>::null_mut
   2: std::panicking::take_hook
   3: std::panicking::take_hook
   4: rustc::ty::structural_impls::<impl rustc::ty::context::Lift<'tcx> for rustc::middle::const_val::ErrKind<'a>>::lift_to_tcx
   5: std::panicking::rust_panic_with_hook
   6: std::panicking::begin_panic_fmt
   7: rust_begin_unwind
   8: core::panicking::panic_fmt
   9: <rustc::util::common::ProfileQueriesMsg as core::fmt::Debug>::fmt
  10: rustc::ty::maps::on_disk_cache::<impl serialize::serialize::SpecializedDecoder<rustc::ty::maps::on_disk_cache::IntEncodedWithFixedSize> for serialize::opaque::Decoder<'enc>>::specialized_decode
  11: rustc::ty::maps::on_disk_cache::<impl serialize::serialize::SpecializedDecoder<rustc::ty::maps::on_disk_cache::IntEncodedWithFixedSize> for serialize::opaque::Decoder<'enc>>::specialized_decode
  12: <rustc::infer::combine::Generalizer<'cx, 'gcx, 'tcx> as rustc::ty::relate::TypeRelation<'cx, 'gcx, 'tcx>>::tys
  13: rustc::ich::caching_codemap_view::CachingCodemapView::byte_pos_to_line_and_col
  14: <rustc::infer::sub::Sub<'combine, 'infcx, 'gcx, 'tcx> as rustc::ty::relate::TypeRelation<'infcx, 'gcx, 'tcx>>::tys
  15: rustc_typeck::hir_trait_to_predicates
  16: rustc_typeck::hir_trait_to_predicates
  17: <rustc::hir::HirId as rustc_typeck::check::writeback::Locatable>::to_span
  18: rustc_typeck::hir_trait_to_predicates
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <rustc_typeck::outlives::implicit_infer::InferVisitor<'cx, 'tcx> as rustc::hir::itemlikevisit::ItemLikeVisitor<'tcx>>::visit_impl_item
  23: <rustc_typeck::outlives::implicit_infer::InferVisitor<'cx, 'tcx> as rustc::hir::itemlikevisit::ItemLikeVisitor<'tcx>>::visit_impl_item
  24: <rustc_typeck::outlives::implicit_infer::InferVisitor<'cx, 'tcx> as rustc::hir::itemlikevisit::ItemLikeVisitor<'tcx>>::visit_impl_item
  25: <rustc_typeck::outlives::implicit_infer::InferVisitor<'cx, 'tcx> as rustc::hir::itemlikevisit::ItemLikeVisitor<'tcx>>::visit_impl_item
  26: <rustc_typeck::check::FnCtxt<'a, 'gcx, 'tcx> as rustc_typeck::astconv::AstConv<'gcx, 'tcx>>::record_ty
  27: <rustc_typeck::check::GatherLocalsVisitor<'a, 'gcx, 'tcx> as rustc::hir::intravisit::Visitor<'gcx>>::visit_pat
  28: <rustc_typeck::collect::has_late_bound_regions::LateBoundRegionsDetector<'a, 'tcx> as rustc::hir::intravisit::Visitor<'tcx>>::visit_lifetime
  29: <rustc_typeck::check::CheckItemTypesVisitor<'a, 'tcx> as rustc::hir::itemlikevisit::ItemLikeVisitor<'tcx>>::visit_item
  30: rustc::ty::maps::<impl rustc::ty::maps::config::QueryConfig<'tcx> for rustc::ty::maps::queries::typeck_tables_of<'tcx>>::compute
  31: rustc::ty::context::tls::track_diagnostic
  32: rustc::dep_graph::graph::DepGraph::assert_ignored
  33: rustc::ty::context::tls::track_diagnostic
  34: rustc::ty::maps::plumbing::<impl rustc::ty::context::TyCtxt<'a, 'gcx, 'tcx>>::try_mark_green_and_read
  35: rustc::ty::maps::plumbing::<impl rustc::ty::context::TyCtxt<'a, 'gcx, 'tcx>>::try_mark_green_and_read
  36: rustc::ty::maps::plumbing::<impl rustc::ty::context::TyCtxt<'a, 'gcx, 'tcx>>::try_mark_green_and_read
  37: <rustc_typeck::check::Diverges as core::fmt::Debug>::fmt
  38: <rustc_typeck::check::CheckItemTypesVisitor<'a, 'tcx> as rustc::hir::itemlikevisit::ItemLikeVisitor<'tcx>>::visit_item
  39: rustc::ty::context::tls::track_diagnostic
  40: rustc::dep_graph::graph::DepGraph::assert_ignored
  41: rustc::ty::context::tls::track_diagnostic
  42: rustc::ty::maps::plumbing::<impl rustc::ty::context::TyCtxt<'a, 'gcx, 'tcx>>::try_mark_green_and_read
  43: rustc::ty::maps::plumbing::<impl rustc::ty::context::TyCtxt<'a, 'gcx, 'tcx>>::try_mark_green_and_read
  44: rustc_typeck::check_crate
  45: <humantime::duration::Error as std::error::Error>::cause
  46: <rustc_driver::pretty::UserIdentifiedItem as core::fmt::Debug>::fmt
  47: <unknown>
  48: rustc_driver::driver::compile_input
  49: rustc_driver::run_compiler
  50: rustc_driver::driver::build_output_filenames
  51: <unknown>
  52: rustc_driver::driver::build_output_filenames
  53: _rust_maybe_catch_panic
  54: rustc_driver::profile::trace::write_style
  55: rustc_driver::main
  56: <unknown>
  57: std::panicking::update_panic_count
  58: _rust_maybe_catch_panic
  59: std::rt::lang_start_internal
  60: <unknown>
  61: <unknown>
  62: BaseThreadInitThunk
  63: RtlUserThreadStart
query stack during panic:
#0 [typeck_tables_of] processing `internal_compiler_errors::no_kind_for_cast`
#1 [typeck_item_bodies] type-checking all item bodies
end of query stack
error: aborting due to previous error

For more information about this error, try `rustc --explain E0019`.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.27.1 (5f2b325f6 2018-07-07) running on x86_64-pc-windows-msvc

note: compiler flags: -C debuginfo=2 -C incremental --crate-type lib

note: some of the compiler flags provided by cargo are hidden

error: Could not compile `test_bugs`.

To learn more, run the command again with --verbose.

```

The backtrace on nightly is verry different:
```
warning: unreachable expression
   --> src\lib.rs:164:25
    |
164 |         let f = [();  { &loop { break } as *const _ as usize } ];
    |                         ^^^^^^^^^^^^^^^
    |
    = note: #[warn(unreachable_code)] on by default

error: internal compiler error: no kind for cast
   --> src\lib.rs:164:25
    |
164 |         let f = [();  { &loop { break } as *const _ as usize } ];
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

error: Could not compile `test_bugs`.

To learn more, run the command again with --verbose.
```

--- a/src/librustc/hir/lowering.rs
+++ b/src/librustc/hir/lowering.rs
@@ -103,7 +103,6 @@ pub struct LoweringContext<'a> {
     loop_scopes: Vec<NodeId>,
     is_in_loop_condition: bool,
     is_in_trait_impl: bool,
-    is_in_anon_const: bool,
 
     /// What to do when we encounter either an "anonymous lifetime
     /// reference". The term "anonymous" is meant to encompass both
@@ -231,7 +230,6 @@ pub fn lower_crate(
         node_id_to_hir_id: IndexVec::new(),
         is_generator: false,
         is_in_trait_impl: false,
-        is_in_anon_const: false,
         lifetimes_to_define: Vec::new(),
         is_collecting_in_band_lifetimes: false,
         in_scope_lifetimes: Vec::new(),
@@ -970,25 +968,21 @@ impl<'a> LoweringContext<'a> {
     }
 
     fn lower_loop_destination(&mut self, destination: Option<(NodeId, Label)>) -> hir::Destination {
-        let target_id = if self.is_in_anon_const {
-            Err(hir::LoopIdError::OutsideLoopScope)
-        } else {
-            match destination {
-                Some((id, _)) => {
-                    if let Def::Label(loop_id) = self.expect_full_def(id) {
-                        Ok(self.lower_node_id(loop_id).node_id)
-                    } else {
-                        Err(hir::LoopIdError::UnresolvedLabel)
-                    }
+        let target_id = match destination {
+            Some((id, _)) => {
+                if let Def::Label(loop_id) = self.expect_full_def(id) {
+                    Ok(self.lower_node_id(loop_id).node_id)
+                } else {
+                    Err(hir::LoopIdError::UnresolvedLabel)
                 }
-                None => {
-                    self.loop_scopes
-                        .last()
-                        .map(|innermost_loop_id| *innermost_loop_id)
-                        .map(|id| Ok(self.lower_node_id(id).node_id))
-                        .unwrap_or(Err(hir::LoopIdError::OutsideLoopScope))
-                        .into()
-                }
+            }
+            None => {
+                self.loop_scopes
+                    .last()
+                    .map(|innermost_loop_id| *innermost_loop_id)
+                    .map(|id| Ok(self.lower_node_id(id).node_id))
+                    .unwrap_or(Err(hir::LoopIdError::OutsideLoopScope))
+                    .into()
             }
         };
         hir::Destination {
@@ -3448,22 +3442,14 @@ impl<'a> LoweringContext<'a> {
     }
 
     fn lower_anon_const(&mut self, c: &AnonConst) -> hir::AnonConst {
-        let was_in_loop_condition = self.is_in_loop_condition;
-        self.is_in_loop_condition = false;
-        let was_in_anon_const = self.is_in_anon_const;
-        self.is_in_anon_const = true;
-
-        let LoweredNodeId { node_id, hir_id } = self.lower_node_id(c.id);
-        let anon_const = hir::AnonConst {
-            id: node_id,
-            hir_id,
-            body: self.lower_body(None, |this| this.lower_expr(&c.value)),
-        };
-
-        self.is_in_anon_const = was_in_anon_const;
-        self.is_in_loop_condition = was_in_loop_condition;
-
-        anon_const
+        self.with_new_scopes(|this| {
+            let LoweredNodeId { node_id, hir_id } = this.lower_node_id(c.id);
+            hir::AnonConst {
+                id: node_id,
+                hir_id,
+                body: this.lower_body(None, |this| this.lower_expr(&c.value)),
+            }
+        })
     }
 
     fn lower_expr(&mut self, e: &Expr) -> hir::Expr {

--- a/src/librustc/session/config.rs
+++ b/src/librustc/session/config.rs
@@ -1193,6 +1193,8 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
           "run all passes except codegen; no output"),
     treat_err_as_bug: bool = (false, parse_bool, [TRACKED],
           "treat all errors that occur as bugs"),
+    report_delayed_bugs: bool = (false, parse_bool, [TRACKED],
+          "immediately print bugs registered with `delay_span_bug`"),
     external_macro_backtrace: bool = (false, parse_bool, [UNTRACKED],
           "show macro backtraces even for non-local macros"),
     teach: bool = (false, parse_bool, [TRACKED],
@@ -3131,6 +3133,10 @@ mod tests {
 
         opts = reference.clone();
         opts.debugging_opts.treat_err_as_bug = true;
+        assert!(reference.dep_tracking_hash() != opts.dep_tracking_hash());
+
+        opts = reference.clone();
+        opts.debugging_opts.report_delayed_bugs = true;
         assert!(reference.dep_tracking_hash() != opts.dep_tracking_hash());
 
         opts = reference.clone();

--- a/src/librustc/session/mod.rs
+++ b/src/librustc/session/mod.rs
@@ -1000,6 +1000,7 @@ pub fn build_session_with_codemap(
     let can_emit_warnings = !(warnings_allow || cap_lints_allow);
 
     let treat_err_as_bug = sopts.debugging_opts.treat_err_as_bug;
+    let report_delayed_bugs = sopts.debugging_opts.report_delayed_bugs;
 
     let external_macro_backtrace = sopts.debugging_opts.external_macro_backtrace;
 
@@ -1045,6 +1046,7 @@ pub fn build_session_with_codemap(
         errors::HandlerFlags {
             can_emit_warnings,
             treat_err_as_bug,
+            report_delayed_bugs,
             external_macro_backtrace,
             ..Default::default()
         },

--- a/src/librustc_errors/diagnostic_builder.rs
+++ b/src/librustc_errors/diagnostic_builder.rs
@@ -132,7 +132,7 @@ impl<'a> DiagnosticBuilder<'a> {
     /// locally in whichever way makes the most sense.
     pub fn delay_as_bug(&mut self) {
         self.level = Level::Bug;
-        *self.handler.delayed_span_bug.borrow_mut() = Some(self.diagnostic.clone());
+        self.handler.delay_as_bug(self.diagnostic.clone());
         self.cancel();
     }
 

--- a/src/librustc_errors/lib.rs
+++ b/src/librustc_errors/lib.rs
@@ -274,7 +274,7 @@ pub struct Handler {
     err_count: AtomicUsize,
     emitter: Lock<Box<dyn Emitter + sync::Send>>,
     continue_after_error: LockCell<bool>,
-    delayed_span_bug: Lock<Option<Diagnostic>>,
+    delayed_span_bug: Lock<Vec<Diagnostic>>,
 
     // This set contains the `DiagnosticId` of all emitted diagnostics to avoid
     // emitting the same diagnostic with extended help (`--teach`) twice, which
@@ -299,7 +299,23 @@ thread_local!(pub static TRACK_DIAGNOSTICS: Cell<fn(&Diagnostic)> =
 pub struct HandlerFlags {
     pub can_emit_warnings: bool,
     pub treat_err_as_bug: bool,
+    pub report_delayed_bugs: bool,
     pub external_macro_backtrace: bool,
+}
+
+impl Drop for Handler {
+    fn drop(&mut self) {
+        if self.err_count() == 0 {
+            let mut bugs = self.delayed_span_bug.borrow_mut();
+            let has_bugs = !bugs.is_empty();
+            for bug in bugs.drain(..) {
+                DiagnosticBuilder::new_diagnostic(self, bug).emit();
+            }
+            if has_bugs {
+                panic!("no errors encountered even though `delay_span_bug` issued");
+            }
+        }
+    }
 }
 
 impl Handler {
@@ -346,7 +362,7 @@ impl Handler {
             err_count: AtomicUsize::new(0),
             emitter: Lock::new(e),
             continue_after_error: LockCell::new(true),
-            delayed_span_bug: Lock::new(None),
+            delayed_span_bug: Lock::new(Vec::new()),
             taught_diagnostics: Lock::new(FxHashSet()),
             emitted_diagnostic_codes: Lock::new(FxHashSet()),
             emitted_diagnostics: Lock::new(FxHashSet()),
@@ -503,11 +519,18 @@ impl Handler {
     }
     pub fn delay_span_bug<S: Into<MultiSpan>>(&self, sp: S, msg: &str) {
         if self.flags.treat_err_as_bug {
+            // FIXME: don't abort here if report_delayed_bugs is off
             self.span_bug(sp, msg);
         }
         let mut diagnostic = Diagnostic::new(Level::Bug, msg);
         diagnostic.set_span(sp.into());
-        *self.delayed_span_bug.borrow_mut() = Some(diagnostic);
+        self.delay_as_bug(diagnostic);
+    }
+    fn delay_as_bug(&self, diagnostic: Diagnostic) {
+        if self.flags.report_delayed_bugs {
+            DiagnosticBuilder::new_diagnostic(self, diagnostic.clone()).emit();
+        }
+        self.delayed_span_bug.borrow_mut().push(diagnostic);
     }
     pub fn span_bug_no_panic<S: Into<MultiSpan>>(&self, sp: S, msg: &str) {
         self.emit(&sp.into(), msg, Bug);
@@ -615,9 +638,6 @@ impl Handler {
 
     pub fn abort_if_errors(&self) {
         if self.err_count() == 0 {
-            if let Some(bug) = self.delayed_span_bug.borrow_mut().take() {
-                DiagnosticBuilder::new_diagnostic(self, bug).emit();
-            }
             return;
         }
         FatalError.raise();

--- a/src/librustc_errors/lib.rs
+++ b/src/librustc_errors/lib.rs
@@ -274,7 +274,7 @@ pub struct Handler {
     err_count: AtomicUsize,
     emitter: Lock<Box<dyn Emitter + sync::Send>>,
     continue_after_error: LockCell<bool>,
-    delayed_span_bug: Lock<Vec<Diagnostic>>,
+    delayed_span_bugs: Lock<Vec<Diagnostic>>,
 
     // This set contains the `DiagnosticId` of all emitted diagnostics to avoid
     // emitting the same diagnostic with extended help (`--teach`) twice, which
@@ -306,7 +306,7 @@ pub struct HandlerFlags {
 impl Drop for Handler {
     fn drop(&mut self) {
         if self.err_count() == 0 {
-            let mut bugs = self.delayed_span_bug.borrow_mut();
+            let mut bugs = self.delayed_span_bugs.borrow_mut();
             let has_bugs = !bugs.is_empty();
             for bug in bugs.drain(..) {
                 DiagnosticBuilder::new_diagnostic(self, bug).emit();
@@ -362,7 +362,7 @@ impl Handler {
             err_count: AtomicUsize::new(0),
             emitter: Lock::new(e),
             continue_after_error: LockCell::new(true),
-            delayed_span_bug: Lock::new(Vec::new()),
+            delayed_span_bugs: Lock::new(Vec::new()),
             taught_diagnostics: Lock::new(FxHashSet()),
             emitted_diagnostic_codes: Lock::new(FxHashSet()),
             emitted_diagnostics: Lock::new(FxHashSet()),
@@ -530,7 +530,7 @@ impl Handler {
         if self.flags.report_delayed_bugs {
             DiagnosticBuilder::new_diagnostic(self, diagnostic.clone()).emit();
         }
-        self.delayed_span_bug.borrow_mut().push(diagnostic);
+        self.delayed_span_bugs.borrow_mut().push(diagnostic);
     }
     pub fn span_bug_no_panic<S: Into<MultiSpan>>(&self, sp: S, msg: &str) {
         self.emit(&sp.into(), msg, Bug);

--- a/src/librustc_resolve/lib.rs
+++ b/src/librustc_resolve/lib.rs
@@ -2358,7 +2358,9 @@ impl<'a> Resolver<'a> {
         where F: FnOnce(&mut Resolver)
     {
         self.ribs[ValueNS].push(Rib::new(ConstantItemRibKind));
+        self.label_ribs.push(Rib::new(ConstantItemRibKind));
         f(self);
+        self.label_ribs.pop();
         self.ribs[ValueNS].pop();
     }
 

--- a/src/librustc_typeck/check/mod.rs
+++ b/src/librustc_typeck/check/mod.rs
@@ -857,9 +857,6 @@ fn typeck_tables_of<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
             fcx.require_type_is_sized(expected_type, body.value.span, traits::ConstSized);
 
             // Gather locals in statics (because of block expressions).
-            // This is technically unnecessary because locals in static items are forbidden,
-            // but prevents type checking from blowing up before const checking can properly
-            // emit an error.
             GatherLocalsVisitor { fcx: &fcx }.visit_body(body);
 
             fcx.check_expr_coercable_to_type(&body.value, expected_type);

--- a/src/librustdoc/core.rs
+++ b/src/librustdoc/core.rs
@@ -161,6 +161,7 @@ pub fn new_handler(error_format: ErrorOutputType, codemap: Option<Lrc<codemap::C
         errors::HandlerFlags {
             can_emit_warnings: true,
             treat_err_as_bug: false,
+            report_delayed_bugs: false,
             external_macro_backtrace: false,
             ..Default::default()
         },

--- a/src/test/ui/const-eval/issue-52442.rs
+++ b/src/test/ui/const-eval/issue-52442.rs
@@ -1,0 +1,13 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+fn main() {
+    [();  { &loop { break } as *const _ as usize } ]; //~ ERROR unimplemented expression type
+}

--- a/src/test/ui/const-eval/issue-52442.stderr
+++ b/src/test/ui/const-eval/issue-52442.stderr
@@ -1,0 +1,9 @@
+error[E0019]: constant contains unimplemented expression type
+  --> $DIR/issue-52442.rs:12:14
+   |
+LL |     [();  { &loop { break } as *const _ as usize } ]; //~ ERROR unimplemented expression type
+   |              ^^^^^^^^^^^^^^
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0019`.

--- a/src/test/ui/const-eval/issue-52443.rs
+++ b/src/test/ui/const-eval/issue-52443.rs
@@ -1,0 +1,18 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+fn main() {
+    [(); & { loop { continue } } ]; //~ ERROR mismatched types
+    [(); loop { break }]; //~ ERROR mismatched types
+    [(); {while true {break}; 0}]; //~ ERROR constant contains unimplemented expression type
+    [(); { for _ in 0usize.. {}; 0}]; //~ ERROR calls in constants are limited to constant functions
+    //~^ ERROR constant contains unimplemented expression type
+    //~| ERROR could not evaluate repeat length
+}

--- a/src/test/ui/const-eval/issue-52443.stderr
+++ b/src/test/ui/const-eval/issue-52443.stderr
@@ -1,0 +1,51 @@
+error[E0308]: mismatched types
+  --> $DIR/issue-52443.rs:12:10
+   |
+LL |     [(); & { loop { continue } } ]; //~ ERROR mismatched types
+   |          ^^^^^^^^^^^^^^^^^^^^^^^
+   |          |
+   |          expected usize, found reference
+   |          help: consider removing the borrow: `{ loop { continue } }`
+   |
+   = note: expected type `usize`
+              found type `&_`
+
+error[E0308]: mismatched types
+  --> $DIR/issue-52443.rs:13:17
+   |
+LL |     [(); loop { break }]; //~ ERROR mismatched types
+   |                 ^^^^^ expected (), found usize
+   |
+   = note: expected type `()`
+              found type `usize`
+
+error[E0019]: constant contains unimplemented expression type
+  --> $DIR/issue-52443.rs:14:11
+   |
+LL |     [(); {while true {break}; 0}]; //~ ERROR constant contains unimplemented expression type
+   |           ^^^^^^^^^^^^^^^^^^
+
+error[E0015]: calls in constants are limited to constant functions, tuple structs and tuple variants
+  --> $DIR/issue-52443.rs:15:21
+   |
+LL |     [(); { for _ in 0usize.. {}; 0}]; //~ ERROR calls in constants are limited to constant functions
+   |                     ^^^^^^^^
+
+error[E0019]: constant contains unimplemented expression type
+  --> $DIR/issue-52443.rs:15:21
+   |
+LL |     [(); { for _ in 0usize.. {}; 0}]; //~ ERROR calls in constants are limited to constant functions
+   |                     ^^^^^^^^
+
+error[E0080]: could not evaluate repeat length
+  --> $DIR/issue-52443.rs:15:10
+   |
+LL |     [(); { for _ in 0usize.. {}; 0}]; //~ ERROR calls in constants are limited to constant functions
+   |          ^^^^^^^^^^^--------^^^^^^^
+   |                     |
+   |                     calling non-const fn `<I as std::iter::IntoIterator><std::ops::RangeFrom<usize>>::into_iter`
+
+error: aborting due to 6 previous errors
+
+Some errors occurred: E0015, E0019, E0080, E0308.
+For more information about an error, try `rustc --explain E0015`.

